Cargo.toml

```
@@ -476,6 +476,34 @@ members = [
 ]
 default-members = ["polkadot", "substrate/bin/node/cli"]
 
+[workspace.lints.rust]
+suspicious_double_ref_op = { level = "allow", priority = 2 }
+
+[workspace.lints.clippy]
+all = { level = "allow", priority = 0 }
+correctness = { level = "deny", priority = 1 }
+if-same-then-else = { level = "allow", priority = 2 }
+complexity = { level = "deny", priority = 1 }
```

Should we set correctness and complexity to warn, since warnings are denied in CI anyway?
```suggestion
correctness = { level = "warn", priority = 1 }
if-same-then-else = { level = "allow", priority = 2 }
complexity = { level = "warn", priority = 1 }
```

---

I think we could but I would suggest to do it as a follow up PR when this is merged,
also there is:
 `borrowed-box = { level = "allow", priority = 2 }                     # Reasonable to fix this one`

---

Fair enough, it'll be nice to have this merged.